src/db/tables/subscriber_email_preferences.ts

```diff
@@ -72,6 +68,38 @@ async function addEmailPreferenceForSubscriber(
   return res?.[0];
 }
 
+// TODO: modify after MNTOR-3557 - pref currently lives in two tables
+// this function only adds unsubscribe token for free reports
+async function addUnsubscribeTokenForSubscriber(
+  subscriberId: number,
+  token: string,
+) {
+  logger.info("add_unsubscribe_token_for_subscriber", {
+    subscriberId,
+  });
+
+  let res;
+  try {
+    res = await knex("subscriber_email_preferences")
+      .insert({
+        subscriber_id: subscriberId,
+        unsubscribe_token: token,
+        monthly_monitor_report_free: null,
+        monthly_monitor_report_free_at: null,
+      })
+      .returning("*");
```

Comment:
@Vinnl wouldn't this suggestion overwrite any existing `unsubscribe_token` for that user? Do we want to preserve old unsubscribe links or is this not a concern here?

Reply:
Yes we want to preserve old unsubscribe links

Reply:
The insert itself would fail if record already exists, it's not meant to be an upsert, just a simple insert. The check is done via a get call before this function is called

Reply:
Ah right, because the subscriber ID is unique, thanks :)

(And @rhelmer, I added the `.whereNull("unsubscribe_token")` to avoid overwriting existing tokens.)